src/app/util/attribute-storage.cpp

```diff
@@ -362,6 +363,57 @@ static void initializeEndpoint(EmberAfDefinedEndpoint * definedEndpoint)
     }
 }
 
+static void shutdownEndpoint(EmberAfDefinedEndpoint * definedEndpoint)
+{
+    // Call shutdown callbacks from clusters, mainly for canceling pending timers
+    uint8_t clusterIndex;
+    const EmberAfEndpointType * epType = definedEndpoint->endpointType;
+    for (clusterIndex = 0; clusterIndex < epType->clusterCount; clusterIndex++)
+    {
+        const EmberAfCluster * cluster = &(epType->cluster[clusterIndex]);
+        EmberAfGenericClusterFunction f;
+        f = emberAfFindClusterFunction(cluster, CLUSTER_MASK_SHUTDOWN_FUNCTION);
+        if (f != nullptr)
+        {
+            ((EmberAfInitFunction) f)(definedEndpoint->endpoint);
```

Comment:
This is a little confusing.  We should have a cast to `EmberAfShutdownFunction` here, and define that typedef in `src/app/util/af-types.h`.

Reply:
done